[MTM-63358] announcement for changes in Web SDK major release (#3222)
SoftwareAG/c8y-docs: content/change-logs/platform-services/ui-c8y-1022-0-0-widget-section-announcement.md

---
date: 2025-04-16
title: Deprecation of loadConfigComponent in favor of new multi-section configuration system
product_area: Application enablement & solutions
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-63358
version: 1022.0.0
---

> **What's changing?**

Reviewer: ```suggestion
**What changes?**
```

> We're introducing a new hookable multi-section concept for widget configuration, and as a result, we are deprecating the loadConfigComponent method of the widgetHook.

Reviewer: ```suggestion
{{< product-c8y-iot >}} introduces a new hookable multi-section concept for the widget configuration, and as a result, the `loadConfigComponent` method of the widgetHook is deprecated.
```

> **What should you do?**

Reviewer: ```suggestion
**How does this change impact you?**
```

> Widget developers should now use the hookWidgetSection hook to add configuration to their own or existing widgets. This new approach supports multiple configuration sections to be displayed, offering greater flexibility and organization of widget settings.

Reviewer: ```suggestion
Widget developers should start using the `hookWidgetSection` hook to add configuration to their own or existing widgets. This new approach supports multiple configuration sections to be displayed, offering greater flexibility and organization of widget settings.
```

**Backward compatibility**

> The loadConfigComponent method will continue to work in the short term and will add a section named "Settings", but it will be removed in an upcoming major release. We recommend migrating to the new system as soon as possible.

Reviewer: ```suggestion

The `loadConfigComponent` method continues to work in short term and will add a section named "Settings", but it will be removed in an upcoming major release. We recommend migrating to the new system as soon as possible.
```

---
date: 2025-04-16

> title: Deprecation of loadConfigComponent in favor of new multi-section configuration system

Reviewer: ```suggestion
title: Deprecation of loadConfigComponent in favor of new multi-section configuration
```

product_area: Application enablement & solutions
change_type:
  - value: change-inv-3bw8e
    label: Announcement
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-63358
version: 1022.0.0
---
**What changes?**

> {{< product-c8y-iot >}} introduces a new hookable multi-section concept for the widget configuration, and as a result, the `loadConfigComponent` method of the widgetHook is deprecated.

Reviewer: ```suggestion

{{< product-c8y-iot >}} introduces a new hookable multi-section concept for the widget configuration, and as a result, the `loadConfigComponent` method of the widgetHook is deprecated.
```

**How does this change impact you?**

> Widget developers should start using the `hookWidgetSection` hook to add configuration to their own or existing widgets. This new approach supports multiple configuration sections to be displayed, offering greater flexibility and organization of widget settings.

Reviewer: ```suggestion

Widget developers should start using the `hookWidgetSection` hook to add configuration to their own or existing widgets. This new approach supports multiple configuration sections to be displayed, offering greater flexibility and organization of widget settings.
```
